tests/test_time_series_graph.py

```diff
@@ -1150,6 +1150,16 @@ def test_get_topological_order(self):
             ['z lag(n=2)', 'x lag(n=2)', 'y lag(n=2)', 'z lag(n=1)', 'x lag(n=1)', 'y lag(n=1)', 'z', 'x', 'y'], order
         )
 
+        tscg = TimeSeriesCausalGraph()
+        # y-1 -> y <- x
+        tscg.add_edge('x', 'y')
+        tscg.add_time_edge('y', -1, 'y', 0)
+
+        order = tscg.get_topological_order(return_all=True)
+        # check there's no duplicates
+        self.assertEqual(len(order), 1)
+        self.assertListEqual(order[0], ['y lag(n=1)', 'x', 'y'])
```

Comment:
Can you add a test where respect time is False and show it is length of 2 and the other valid one is there?